### DISCUSSION TITLE
分享一个绑定调试方法,用于处理绑定异常.

### DISCUSSION THREAD

**Discussion by xingzuhui** [BYJRK/DotNet-Discussions#205] [WPF 开发]
wpf绑定数据的时候,经常遇到不起作用,又不知道是绑定的方法签名错了,属性错了,数据类型错了还是转换错了.
怎么办呢?就需要调试,wpf有没法跟踪断点,或者说我还不会跟踪wpf断点. 那么就采用这中方法:
在窗口中添加
`xmlns:trace="clr-namespace:System.Diagnostics;assembly=WindowsBase"`
在需要跟踪的绑定上
`   <TextBlock FontSize="48" HorizontalAlignment="Center" VerticalAlignment="Center" Text="{Binding RelativeSource={RelativeSource AncestorType=ListBoxItem}, Path=(ItemsControl.AlternationIndex),trace:PresentationTraceSources.TraceLevel=High, Mode=OneWay, Converter={StaticResource NumberAddOne}}">`

**Comment by BYJRK** [comment]
这个有点意思，明天试一下。其实一般的绑定问题，从 VS 自带的那个 XAML 绑定失败里面就基本上能看出来了
